src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,219 @@
+    PROPERTY_MAPPER.put("tax", Param.TAX);
+    PROPERTY_MAPPER.put("total", Param.VALUE);
+    PROPERTY_MAPPER.put("revenue", Param.VALUE);
+    PROPERTY_MAPPER.put("order_id", Param.TRANSACTION_ID);
+    PROPERTY_MAPPER.put("currency", Param.CURRENCY);
+    return PROPERTY_MAPPER;
   }
 
-  @Override public void onActivityStopped(Activity activity) {
-    super.onActivityStopped(activity);
+  public FirebaseIntegration(Context context, Logger logger) {
+    this.logger = logger;
+    firebaseAnalytics = FirebaseAnalytics.getInstance(context);
+  }
 
+  @Override
+  public void onActivityResumed(Activity activity) {
+    super.onActivityResumed(activity);
+
+    PackageManager packageManager = activity.getPackageManager();
+    try {
+      ActivityInfo info =
+          packageManager.getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);
+      String activityLabel = info.loadLabel(packageManager).toString();
+      firebaseAnalytics.setCurrentScreen(activity, activityLabel, null);
+      logger.verbose(
+          "firebaseAnalytics.setCurrentScreen(activity, %s, null);",
+          activityLabel);
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError("Activity Not Found: " + e.toString());
+    }
   }
 
-  @Override public void identify(IdentifyPayload identify) {
+  @Override
+  public void identify(IdentifyPayload identify) {
     super.identify(identify);
 
     if (!isNullOrEmpty(identify.userId())) {
-      mFirebaseAnalytics.setUserId(identify.userId());
+      firebaseAnalytics.setUserId(identify.userId());
     }
+    Map<String, Object> traits = identify.traits();
+    for (Map.Entry<String, Object> entry : traits.entrySet()) {
+      String trait = entry.getKey();
+      Object value = entry.getValue();
+      trait = makeKey(trait);
+      String formattedValue;
+      if (value instanceof Date) {
+        Date dateValue = (Date) value;
+        formattedValue = FIREBASE_FORMAT.format(dateValue);
+      } else {
+        formattedValue = String.valueOf(value);
+      }
+      firebaseAnalytics.setUserProperty(trait, formattedValue);
+      logger.verbose("firebaseAnalytics.setUserProperty(%s, %s);", trait, formattedValue);
+    }
+  }
 
-    // mFirebaseAnalytics.setUserProperty(key, value);
+  @Override
+  public void track(TrackPayload track) {
+    super.track(track);
 
+    String event = track.event();
+    String eventName = makeKey(event);
+    Properties properties = track.properties();
+    Bundle formattedProperties = formatProperties(properties);
+    firebaseAnalytics.logEvent(eventName, formattedProperties);
+    logger.verbose("firebaseAnalytics.logEvent(%s, %s);", eventName, formattedProperties);
   }
 
-  @Override public void screen(ScreenPayload screen) {
-    super.screen(screen);
-//    mFirebaseAnalytics.setCurrentScreen(this, screen.name(), null /* class override */);
+  private Bundle formatProperties(Properties properties) {
```

Comment:
once the logger is removed from this function, this should be a static function.